### PR TITLE
feat: add support for entry and asset hyperlinks

### PR DESCRIPTION
**Pull request by cezarsmpio**
# Entry

image
image

# Asset

image
image

### CODE REVIEW

**Review by taye — APPROVED** [review]

File: packages/rich-text/src/helpers/editor.ts [hunk 1]
```diff
@@ -124,25 +129,31 @@ export function unwrapLink(editor) {
...
-    data: {
-      uri: url,
-    },
+    type,
+    data: {},
     children: isCollapsed ? [{ text }] : [],
   };
 
+  if (url) {
+    link.data = { url };
```
Comment (cribbles) [inline packages/rich-text/src/helpers/editor.ts:150]: It's okay that this is `url` and not `uri: url`?

File: packages/rich-text/src/helpers/editor.ts [hunk 2]
```diff
@@ -124,25 +129,31 @@ export function unwrapLink(editor) {
...
+  }
+
+  if (target) {
+    link.data = { target };
+  }
```
Comment (cribbles) [inline packages/rich-text/src/helpers/editor.ts:155]: Would we face any issues with a scenario where `url` and `target` are both present?

File: packages/rich-text/src/plugins/Hr/index.tsx [hunk 3]
```diff
@@ -101,6 +102,8 @@ export function ToolbarHrButton(props: ToolbarHrButtonProps) {
     Slate.ReactEditor.focus(editor);
   }
 
+  if (!editor) return null;
```
Comment (cribbles) [inline packages/rich-text/src/plugins/Hr/index.tsx:105]: What are these changes for?

File: packages/rich-text/src/helpers/editor.ts (hunk 1 above)
Comment (cezarsmpio) [reply packages/rich-text/src/helpers/editor.ts:150]: Good catch!

File: packages/rich-text/src/plugins/List/index.tsx [hunk 4]
```diff
@@ -30,6 +30,8 @@ export function ToolbarListButton(props: ToolbarListButtonProps) {
     Slate.ReactEditor.focus(editor);
   }
 
+  if (!editor) return null;
```
Comment (cribbles) [inline packages/rich-text/src/plugins/List/index.tsx:33]: same here, what does this fix?

File: packages/rich-text/src/plugins/Hr/index.tsx (hunk 3 above)
Comment (cezarsmpio) [reply packages/rich-text/src/plugins/Hr/index.tsx:105]: Lint was accusing some errors when I was trying to improve the internal type. `editor` might be undefined. We also used the same pattern on other buttons but forgot to add on some. I can remove it if that's causing noise and confusion.

File: packages/rich-text/src/plugins/List/index.tsx (hunk 4 above)
Comment (cezarsmpio) [reply packages/rich-text/src/plugins/List/index.tsx:33]: Answer above :)

File: packages/rich-text/src/plugins/Hr/index.tsx (hunk 3 above)
Comment (cribbles) [reply packages/rich-text/src/plugins/Hr/index.tsx:105]: 👍

Comment (cribbles) [reply packages/rich-text/src/plugins/Hr/index.tsx:105]: (👍 to keeping it, not removing)

File: packages/rich-text/src/helpers/editor.ts (hunk 2 above)
Comment (cezarsmpio) [reply packages/rich-text/src/helpers/editor.ts:155]: I don't think we can both at the same type. Nevertheless, `type` will dictate what to render, and the render will either use `target` or `uri`.

**Pull request merged**